libs/core/src/main/proto/tasks/TaskIdProto.proto

```diff
@@ -0,0 +1,20 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+syntax = "proto3";
+package org.opensearch.core.tasks.proto;
+
+option java_outer_classname = "TaskIdProto";
+
+message TaskId {
+  string nodeId = 1;
+  int64 id = 2;
```

Comment:
If you choose to serialize the task ID instead of using the "0" byte, consider making these `optional` parts of the message with the defaults from the empty string/-1, to accomplish the same behavior but permit serializing "nothing" rather than a string and int64 (I'm curious what a "nothing" byte stream looks like.)

Reply:
That makes sense! We can make the `nodeId` optional in the proto message.

Reply:
> We can make the nodeId optional in the proto message.

Note this is an alternative to the "0" shortcut which I think is a better option.   Is it allowed to have all the values optional?  What is the byte size of a protobuffer with a missing nodeId and a single -1 long, which would be the constant we'd store and use as an alternative to "0"?